src/app/Plans/components/helpers.ts

```diff
@@ -1,122 +1,175 @@
+  let title = '';
+  let variant: ProgressVariant | undefined;
+
+  switch (true) {
+    case state === 'Starting': {
+      title = 'Running';
+      break;
+    }
+    case state === 'NotStarted': {
+      title = '';
+      break;
+    }
+    case state === 'Finished-Failed':
+    case state === 'AbortedCopying': {
+      title = 'Failed';
+      variant = ProgressVariant.danger;
+      break;
+    }
+    case state === 'Finished-Succeeded': {
+      title = 'Succeeded';
+      variant = ProgressVariant.success;
+      break;
+    }
+    case state === 'Copying':
+    case state === 'PipelineRunning': {
+      title = isWarmPlan ? 'Running cutover' : 'Running';
+      break;
+    }
+    case state === 'Finished-Incomplete': {
+      title = 'Finished - Incomplete';
+      variant = ProgressVariant.warning;
+      break;
+    }
+    default: {
+      title = '';
+    }
+  }
+
+  return {
+    title,
+    variant,
+  };
+};
+
+export const getButtonState = (state: PlanState | null): PlanActionButtonType | null => {
+  let type: PlanActionButtonType | null = null;
+
+  switch (true) {
+    case state === 'NotStarted': {
+      type = 'Start';
+      break;
+    }
+    case state === 'Starting': {
+      type = null;
+      break;
+    }
+    case state === 'Copying': {
+      type = 'Cutover';
+      break;
+    }
+    case state === 'Finished-Incomplete':
+    case state === 'Finished-Failed':
+    case state === 'Canceled':
+    case state === 'AbortedCopying': {
+      type = 'Restart';
+      break;
+    }
+    case state === 'StartingCutover': {
+      type = null;
+      break;
+    }
+    case state === 'PipelineRunning': {
+      type = null;
+      break;
+    }
+    case state === 'Finished-Succeeded': {
+      type = null;
+      break;
+    }
+    default: {
+      type = null;
+    }
+  }
+
+  return type;
+};
+
+export const getPlanState = (
   plan: IPlan | null,
-  migration: IMigration | null,
-  migrationsQuery: UseQueryResult<IKubeList<IMigration>>
-): WarmPlanState | null => {
-  if (!plan || !plan.spec.warm) return null;
+  migration: IMigration | null
+): PlanState | null => {
+  if (!plan) return null;
+
   if (!migration) return 'NotStarted';
-  if (isPlanBeingStarted(plan, migration, migrationsQuery)) {
-    return 'Starting';
-  }
+
   const conditions = plan.status?.conditions || [];
+
   if (
-    (hasCondition(conditions, PlanStatusType.Canceled) ||
-      hasCondition(conditions, PlanStatusType.Failed)) &&
+    (hasCondition(conditions, 'Canceled') || hasCondition(conditions, 'Failed')) &&
     !migration.spec.cutover
   ) {
     return 'AbortedCopying';
   }
-  if (!!migration && !!plan.status?.migration?.completed) return 'Finished';
-  if (hasCondition(conditions, PlanStatusType.Executing)) {
+
+  if (!!plan.status?.migration?.started && canPlanBeStarted(plan)) {
+    return 'Finished-Incomplete';
+  }
+
+  if (!!migration && !!plan.status?.migration?.completed) {
```

Comment:
This logic will return Finished-Succeeded if a plan has a `completed` timestamp, but plans get that timestamp even if they are failed or canceled. `plan.status?.migration?.completed` just means the plan is no longer running. It may be simplest to move your Canceled and Finished-Failed cases from line 152-159 to above this code so the rest of the logic can assume there was no failure.

Wait, hang on... you have two cases that return Finished-Succeeded? I think if you move all 3 of the cases from the bottom (Finished-Succeeded, Canceled, Finished-Failed) above this one you can delete this one.